router/realm.go

```diff
@@ -291,6 +309,27 @@ func (r *realm) onLeave(sess *wamp.Session, shutdown bool) {
 	<-sync
 
 	if !shutdown {
+		if testaments, ok := r.testaments[sess.ID]; ok {
+			for _, testament := range testaments.detached {
+				r.metaPeer.Send(&wamp.Publish{
+					Request:     wamp.GlobalID(),
+					Topic:       testament.topic,
+					Arguments:   testament.args,
+					ArgumentsKw: testament.kwargs,
+					Options:     testament.options,
+				})
+			}
+			for _, testament := range testaments.destroyed {
+				r.metaPeer.Send(&wamp.Publish{
+					Request:     wamp.GlobalID(),
+					Topic:       testament.topic,
+					Arguments:   testament.args,
+					ArgumentsKw: testament.kwargs,
+					Options:     testament.options,
+				})
+			}
+			delete(r.testaments, sess.ID)
+		}
```

Comment:
Since `testaments.detached` and `testaments.destroyed` are not slices of pointers, the loops above will copy the testament struct each iteration.  Instead, use an index to access each slice element.  Indexing within a loop is highly optimized in go. Also, since the loops are identical, except for the slice they iterate, use a function so that the code does not repeat:
```go
sendTestaments = func(testaments []testament) {
    for i := range testaments {
        r.metaPeer.Send(&wamp.Publish{
            Request:     wamp.GlobalID(),
            Topic:       testaments[i].topic,
            Arguments:   testaments[i].args,
            ArgumentsKw: testaments[i].kwargs,
            Options:     testaments[i].options,
        })
    }
}
sendTestaments(testaments.detached)
sendTestaments(testaments.destroyed)
```